src/lib.rs

```diff
@@ -273,8 +273,11 @@ impl LsColors {
             if let Some([entry, ansi_style]) = parts.get(0..2) {
                 let style = Style::from_ansi_sequence(ansi_style);
                 if let Some(suffix) = entry.strip_prefix('*') {
-                    self.suffix_mapping
-                        .push((suffix.to_string().to_ascii_lowercase(), style));
+                    // in LS the latest entry takes precedence
+                    // hashmap allows us to overwrite already present entries with updated values
+                    // thus taking the latest value given
+                    // self.suffix_mapping.retain(|&k, _| { k.ends_with(suffix.to_string()) });
```

Comment:
I think you want something like

```rust
self.suffix_mapping.retain(|k, _v| !k.ends_with(&suffix));
```

Reply:
279 is redundant i think, it was from an earlier try. But using the suffix filter here would circumvent all logic. I'll update it.